### ISSUE TITLE
Generate exchange lines: use spatial indexing TOPDESK 2402 0060

### ISSUE THREAD

**Issue by leendertvanwolfswinkel** [nens/threedi-schematisation-editor#217]

**Comment by ldebek** [comment]
@leendertvanwolfswinkel @Overmeen  Looked through the algorithm code, and it's not clear for me on which alg step shall I utilise spatial indexing? Could you please provide more details to this ticket?

**Comment by leendertvanwolfswinkel** [comment]
@ldebek yeah I looked through the description of the algorithm again and no spatial predicate is used anywhere, so it doesn't make much sense indeed.

The reason for this ticket is that some users complain that the algorithm is slow. Would there be ways to speed up the algorithm? E.g. generate the offset curves in memory and add them to the Exchange line layer all at once at the end? Or other changes?
